app/code/Magento/Catalog/Controller/Category/View.php

```diff
@@ -210,6 +209,9 @@ public function execute()
             return $this->resultRedirectFactory->create()->setUrl($this->_redirect->getRedirectUrl());
         }
         $category = $this->_initCategory();
+        if ($this->toolbarMemorizer->hasMemorizingParam()) {
```

Comment:
You should also check if memorizing is allowed via $this->toolbarMemorizer->isMemorizingAllowed() method call.

Reply:
I suspect it is necessary. You can implement the change and have the tests run again and see how it goes. We can always revert the change. Please bare in mind that in order for this to pass, all tests need to pass. 

Also, once we have a stable solution, we will need to have testing coverage for the new code.